script.rb

```diff
@@ -0,0 +1,42 @@
+require 'csv'
+require "google_drive"
+require 'mechanize'
+require 'rails'
+require 'byebug'
+
+
+id_array = ['ID']
+
+id_array.each do |id|
+
+  pass, sp_url = 'PASS', "https://docs.google.com/spreadsheets/d/1mNAJZp0B0nLZ6dYuoMZuYDmIRUd_w8PiiY_NtEUFTW8/edit#gid=0" if id == 'ID'
+
+  start_date = (Time.now.beginning_of_month).to_i.to_s + "000"
+  end_date = (Time.now).to_i.to_s + "999"
+  agent = Mechanize.new
+  page = agent.get('https://twitter.com/')
+  form = page.forms[1]
+  form.field_with(name: "session[username_or_email]").value = id
+  form.field_with(name: "session[password]").value = pass
+  form.submit
+  export_url = "https://analytics.twitter.com/user/" + id + "/tweets/export.json?start_time=" + start_date + "&end_time=" + end_date + "&lang=ja"
+  bundle_url = "https://analytics.twitter.com/user/" + id + "/tweets/bundle?start_time=" + start_date + "&end_time=" + end_date + "&lang=ja"
+  agent.post(export_url)
+  file = agent.get(bundle_url)
+  file.save
```

Comment:
I save file to avoid following error.

```
*** Errno::ENOENT Exception: No such file or directory @ rb_sysopen - tweet_activity_metrics_Playerapp_vb_20180731_20180830_ja.csv
```

Reply:
https://docs.ruby-lang.org/ja/latest/class/Encoding.html この辺でなんとかなるはず

Reply:
ご参考に

```ruby
res = agent.get(bundle_url)
csv = res.body.force_encoding('utf-8')
```